src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+[assembly: System.Resources.NeutralResourcesLanguage("en-us")]
+
+namespace System.Text.RegularExpressions.Generator
+{
+    /// <summary>Generates C# source code to implement regular expressions.</summary>
+    [Generator(LanguageNames.CSharp)]
+    public partial class RegexGenerator : IIncrementalGenerator
+    {
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            IncrementalValueProvider<ImmutableArray<object?>> typesAndCompilation =
+                context.SyntaxProvider
+
+                // Find all MethodDeclarationSyntax nodes attributed with RegexGenerator
+                .CreateSyntaxProvider(static (s, _) => IsSyntaxTargetForGeneration(s), static (ctx, _) => GetSemanticTargetForGeneration(ctx))
+                .Where(static m => m is not null)
+
+                // Pair each with the compilation
+                .Combine(context.CompilationProvider)
+
+                // Use a custom comparer that ignores the compilation so that it doesn't interface with the generators caching of results based on MethodDeclarationSyntax
```

Comment:
From reading this comment, I'm not sure I understand what is happening here. (Even if I replace `interface` => `interfere` I still don't think I get why this is necessary.)

Reply:
Should have said interfere.  The idea is that we want to avoid regenerating code for a regex unless that regex has been changed.  Roslyn caches outputs based on SyntaxNodes, so as long as we can arrange things such that we produce output per SyntaxNode, we can make it so that we don't need to recompute for SyntaxNodes that haven't changed.  But changing any SyntaxNode will also change the Compilation, so we need to ignore the Compilation as part of the comparison operation Roslyn will do to determine whether things need to be recomputed.

Reply:
Is this purely a perf optimization?

Reply:
To avoid re-doing unnecessary work,  yes.